core/DataAccess/ArchiveSelector.php

```diff
@@ -542,7 +542,7 @@ private static function findArchiveDataWithLatestTsArchived($results, $requested
                 }
 
                 if (empty($archiveData[$metric])) {
-                    if (!empty($result[$metric]) || $result[$metric] === 0 || $result[$metric] === '0') {
+                    if (!empty($result[$metric]) || $result[$metric] === 0 || $result[$metric] === 0.0 || $result[$metric] === '0') {
```

Comment:
Since we're updating this, I wonder if we could make it a bit shorter and better understandable than checking for all different kinds of zero
```suggestion
                    if (!empty($result[$metric]) || (is_numeric($result[$metric]) && $result[$metric] == 0)) {
```

Just a thought, no need to update if you don't find that better.

Reply:
Hm. your suggestion looks shorter. However I think we should overall try to use more type specific compares, rather than such loose ones. There were already a bunch of problems with PHP 8, due to some changes how such compares are handled. So I think it's better to be very explicit with what we expect.